### ISSUE TITLE
SVGGradientLayer & -renderInContext Fixable!

### ISSUE THREAD

**Issue by DigitalExegete** [SVGKit/SVGKit#262]
I'm using the SVGKit to render some images, and I found out that if you add these lines to the to the beginning & end of -renderInContext:, the Gradients are rendered properly:

At the beginning:

```
if (self.mask)
{
    self.layerMask = self.mask;
    self.mask = nil;
}
```

At the end:

```
if (self.layerMask)
{
    self.mask = self.layerMask;
    self.layerMask = nil;
}
```

Also, you would need make a class extension and add the following

```
 @property (strong) CALayer *layerMask;
```

I would just make a pull request, but I have forked from MaddTheSane's fork.

![screen shot 2015-01-25 at 14 27 37](https://cloud.githubusercontent.com/assets/7121337/5893144/7515c964-a49e-11e4-84ab-15c3f73e3f10.png)

**Comment by adamgit** [comment]
So ... you're:
1. Removing the mask
2. Applying the mask to the main object after you've rendered

?

I don't understand how this works / could work - can you explain? It seems
like it would only succeed as an accident based on you ALSO rendering the
object WITHOUT renderInContext (but I'm probably misunderstanding it)

**Comment by DigitalExegete** [comment]
Yes, 1 & 2.  You still need the mask when you are displaying the layer on screen, but when it comes time to render, you have to remove the layer's mask property.  Where I am using it (OS X), I am using an NSView to show users various things, and then I use that same view to render to a PDFContext (to print).

The thing with CALayers is that their masks operate by masking based on the alpha value, so 0% opacity for a masking layer lets nothing through, and 100% opacity lets everything through (which seems backwards).  However, when the a UIView/NSView is performing renderInContext, this is ignored, and the mask is rendered as a normal layer.  So, removing the masking layer, and using the clip path seems to work.

**Comment by DigitalExegete** [comment]
Actually, it appears that on OS X, the mask property is rendered incorrectly, but on iOS, it is (now) treated correctly.  

I cleared the layer's mask property, and let the clip path do the masking, and it rendered correctly. I then commented out the clip path, and left the masking layer, and it also rendered correctly.  Just to be sure, I cleared both, and the gradient rendered as a big rectangle.  

It looks like Apple fixed -renderInContext  with CAGradientLayers on iOS 8.1.  However, OS X remains a different story, as the masking layer is added as a normal layer.
